src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.Shared.xml

```diff
@@ -56,5 +56,9 @@
       <!-- Methods is used by VS Tasks Window. -->
       <method name="GetActiveTaskFromId" />
     </type>
+    <!-- Accessed via private reflection by an external tracing controller. -->
+    <type fullname="System.Diagnostics.Tracing.EventPipe*" />
```

Comment:
This should be under `System.Diagnostics.Tracing.EventSource.IsSupported` feature

/cc @eerhardt

Reply:
Like this:

<type fullname="System.Diagnostics.Tracing.EventPipe*" feature="System.Diagnostics.Tracing.EventSource.IsSupported" featurevalue="true" />

?

Reply:
This needs to be moved out to its own `assembly` section because the existing one is already conditional. Something like

```xml
<assembly fullname="System.Private.CoreLib" feature="System.Diagnostics.Tracing.EventSource.IsSupported" featurevalue="true" featuredefault="true">
	<type fullname="System.Diagnostics.Tracing.EventPipe*" />
</assembly>

Reply:
This isn't the right fix. The reason this entry is in CoreCLR is because there is a profiling service (i.e. the "external tracing controller" mentioned in the comments) calls these methods/classes using private reflection. This is only used on ASP.NET services.

So instead, we shouldn't be rooting all `EventPipe*` on mono - since this isn't necessary. Instead, we should inspect what is actually needed, and only root that (under the EventSource.IsSupported feature switch of course).

Reply:
Also note that I've requested the profiling service team to remove this private reflection and instead use the public NuGet package that does the same thing: https://www.nuget.org/packages/Microsoft.Diagnostics.NETCore.Client

cc @xiaomi7732 @cweining @noahfalk

Reply:
Note that there are also a number of CoreCLR tests that relies on a copuple of EventPipe classes using private reflection, runtime\src\tests\tracing\common\TraceConfiguration.cs, runtime\src\tests\tracing\common\TraceControl.cs

Reply:
So for this test, IEventProvider.EventActivityIdControl and EventPipeEventProvider.IEventProvider.EventActivityIdControl gets eliminated and needs to be preserved. So in what file should I put that based on above discussions?

Reply:
Can we fix the test to not use private reflection? That seems the most appropriate fix to me.

Reply:
Several of the CoreCLR diagnostics tests uses this pattern to test internal classes and methods, currently not exposed outside S.P.C.

Reply:
Can we use the code from https://www.nuget.org/packages/Microsoft.Diagnostics.NETCore.Client instead of using private reflection? Or basically have some test code that enables the event pipes without using reflection.

Reply:
These are low level tests, just testing basic internal mechanism of implementation (in this case just EventPipeInternal.EventActivityIdControl). Do we have any existing mechanism in S.P.C to include code only when running tests?